pkg/proxy/proxier.go

```diff
@@ -665,6 +725,18 @@ func iptablesInit(ipt iptables.Interface) error {
 	if _, err := ipt.EnsureRule(iptables.TableNAT, iptables.ChainOutput, "-j", string(iptablesHostPortalChain)); err != nil {
 		return err
 	}
+	if _, err := ipt.EnsureChain(iptables.TableNAT, iptablesContainerPublicChain); err != nil {
+		return err
+	}
+	if _, err := ipt.EnsureRule(iptables.TableNAT, iptables.ChainPrerouting, "-j", string(iptablesContainerPublicChain)); err != nil {
+		return err
+	}
+	if _, err := ipt.EnsureChain(iptables.TableNAT, iptablesHostPublicChain); err != nil {
+		return err
+	}
+	if _, err := ipt.EnsureRule(iptables.TableNAT, iptables.ChainOutput, "-j", string(iptablesHostPublicChain)); err != nil {
```

Comment:
This is a little scary - if somehow these rules get out of order, this will consume all accesses to a given port number regardless of destIP.  Should this maybe use -m addrtype --dst-type LOCAL before -j ?

And I think this should be installed in the OUTPUT chain too, no?  PREROUTING is from off-machine or from containers.  OUTPUT is from the node itself.  I think that should work.